### PR TITLE
fix: skip paths based on flag

### PR DESCRIPTION
**Pull request by shrutiburman**
# Fixes #
There are some paths that need not be present in SDKs generation. These are flagged with `x-skip-operation: 'true'`. Such as /healthcheck apis, these are essentially internal APIs but the status-check pipeline also [validates](https://buildkite.com/twilio/twilio-librarian-status-checks/builds/6102#018c6149-0ff3-46bf-95a2-228b13ffc592/3475-3510) the private visibility apis, so the checks are failing. 
To process these flags the changes are needed in oai-generator and transpiler. 

Related PR: https://code.hq.twilio.com/dx/open-api-transpiler/pull/7
### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] Run `make test-docker`
- [ ] Verify affected language:
    - [ ] Generate [twilio-go](https://github.com/twilio/twilio-go) from our [OpenAPI specification](https://github.com/twilio/twilio-oai) using the [build_twilio_go.py](./examples/build_twilio_go.py) using `python examples/build_twilio_go.py path/to/twilio-oai/spec/yaml path/to/twilio-go` and inspect the diff
    - [ ] Run `make test` in `twilio-go`
    - [ ] Create a pull request in `twilio-go`
    - [ ] Provide a link below to the pull request
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-oai-generator/blob/main/CONTRIBUTING.md) and my PR follows them
- [ ] I have titled the PR appropriately
- [ ] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please create a GitHub Issue in this repository.

### CODE REVIEW

File: src/main/java/com/twilio/oai/DirectoryStructureService.java [hunk 1]
```diff
@@ -72,11 +72,17 @@ public static class ContextResource {
...
         final Map<String, DependentResource> versionResources = getVersionResourcesMap();
+        Map<String, PathItem> pathsToSkipMap = new HashMap<>();
 
         isVersionLess = additionalProperties.getOrDefault(API_VERSION, "").equals("");
 
         openAPI.getPaths().forEach(resourceTree::addResource);
         openAPI.getPaths().forEach((name, path) -> {
+            final Optional<String> skipPath = Optional.of(
+                // if the flag is set, this path need not be generated by helper libraries
+                PathUtils.getTwilioExtension(path, "x-skip-operation").orElse(""));
```
Comment (sbansla) [inline src/main/java/com/twilio/oai/DirectoryStructureService.java:83]: if` x-skip-operation` is marked as false. Will it skip or keep the path ?

Comment (shrutiburman) [reply src/main/java/com/twilio/oai/DirectoryStructureService.java:83]: It'll keep the path, work as usual.

File: examples/spec/twilio_api_v2010.yaml [hunk 2]
```diff
@@ -554,3 +554,46 @@ paths:
       mountName: feedback_call_summary
       parent: /Accounts/{AccountSid}/Calls/{TestInteger}.json
       pathType: instance
+  /healthcheck:
+    x-twilio:
+      className: healthCheck
+      libraryVisibility: private
+      docs_visibility: private
+      pathType: list
+      x-skip-operation: 'true'
```
[the remark this answers is not in the document]
Comment (shrutiburman) [reply examples/spec/twilio_api_v2010.yaml:563]: Yes, this works at path level. I think I should rename it as `x-skip-path` soon.

**Comment by shrutiburman** [comment]
https://code.hq.twilio.com/dx/open-api-transpiler/pull/7

**Review by sbansla — APPROVED** [review]

**Pull request merged**